src/oemof/solph/components/_generic_storage.py

```diff
@@ -421,70 +427,29 @@ def _storage_content_bound_rule(block, n, t):
             return bounds
 
         self.storage_content = Var(
-            self.STORAGES, m.TIMESTEPS, bounds=_storage_content_bound_rule
-        )
-
-        def _storage_init_content_bound_rule(block, n):
-            return 0, n.nominal_storage_capacity
-
-        self.init_content = Var(
-            self.STORAGES,
-            within=NonNegativeReals,
-            bounds=_storage_init_content_bound_rule,
+            self.STORAGES, m.TIMEPOINTS, bounds=_storage_content_bound_rule
         )
 
         # set the initial storage content
+        # ToDo: More elegant code possible?
         for n in group:
             if n.initial_storage_level is not None:
-                self.init_content[n] = (
+                self.storage_content[n, 0] = (
                     n.initial_storage_level * n.nominal_storage_capacity
                 )
-                self.init_content[n].fix()
+                self.storage_content[n, 0].fix()
```

Comment:
I know, I had a different argumentation on this, but I spoke again to a colleague of mine:
* It seems not so uncommon to do something like it originally was, i.e. to define the first storage transition (from t=-1 to t= 0) separately. This allows to include some freedom to the model instead of just fixing the first variable value (for t=0) as I suggested in the discussion.
* You basically changed the definition of the `storage_content` from after inflows and outflows to before inflows and outflows of the respective timestep. Thus, the storage level for t=0 is now fixed, whereas you add |T|+1. When you cut this last time point in results processing, you loose an information actually determined by the model in favor of some information you specified externally.

Anyways, it remains a matter of definition. I also see not very much of a problem for modelers (like me) that think rather in the same intervals for all vars instead of disctincting average values for time intervals (for flows) and discrete time points (for storages). Either way, you have to drop one value if you think this way.

A side note: If you want to retain the original behaviour - which it seems we don't, actually, TIMEPOINTS[2] results would correspond with TIMESTEPS[1].